Add PopcornTime to Flathub!

Preisschild opened the pull request:
Adds PopcornTime Desktop: https://github.com/popcorn-official/popcorn-desktop

TingPing commented on sh.popcorntime.PopcornTime.yml:
```diff
@@ -0,0 +1,42 @@
+runtime-version: '18.08'
+branch: stable
+sdk: org.freedesktop.Sdk
+base: org.electronjs.Electron2.BaseApp
```
```suggestion
base: io.atom.electron.BaseApp
```

TingPing commented on sh.popcorntime.PopcornTime.yml:
```diff
@@ -0,0 +1,42 @@
+branch: stable
+sdk: org.freedesktop.Sdk
+base: org.electronjs.Electron2.BaseApp
+base-version: stable
```
```suggestion
base-version: '18.08'
```

TingPing commented on sh.popcorntime.PopcornTime.yml:
```diff
@@ -0,0 +1,42 @@
+- --filesystem=home:ro
+- --talk-name=org.freedesktop.Notifications
+modules:
+- shared-modules/gtk2/gtk2.json
```
```suggestion
```

Comes with correct baseapp.

x80486 commented on sh.popcorntime.PopcornTime.appdata.xml:
```diff
@@ -0,0 +1,49 @@
+    <screenshot>https://i.imgur.com/1NrtHBj.png</screenshot>    
+  </screenshots>
+  <releases>
+    <release version="0.3.10" date="2018-05-14"/>
```
`0.3.10` was released in Oct 31, 2016 ...not a few months ago.

Preisschild replied:
The version from their CI (https://ci.popcorntime.sh/job/Popcorn-Time-Desktop/) has a few fixes and was released a few months ago. Since they have not changed the version, I went with 0.3.10, but I'm open for input to change the version number.

bilelmoussaoui commented:
The app works on all arches? If not you should add a flathub.json file to specify which arch should be built.

TingPing commented on sh.popcorntime.PopcornTime.appdata.xml:
```diff
@@ -0,0 +1,49 @@
+<!-- Copyright 2018 Ströger Florian -->
+<?xml version="1.0" encoding="UTF-8"?>
+<component type="desktop">
+  <id>sh.popcorntime.PopcornTime.desktop</id>
```
(It doesn't cause trouble but it is the older way of doing it)

Preisschild commented:
Added all your suggestions and removed the now-useless shared-modules submodule. This is my first time packaging apps to flatpak, so thanks for bearing with me!

TingPing commented on sh.popcorntime.PopcornTime.appdata.xml:
```diff
@@ -0,0 +1,50 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<component type="desktop">
+  <id>sh.popcorntime.PopcornTime</id>
+  <launchable type="sh.popcorntime.PopcornTime.desktop"/>
```
```suggestion
  <launchable type="desktop-id">sh.popcorntime.PopcornTime.desktop</launchable>
```

TingPing commented on sh.popcorntime.PopcornTime.appdata.xml:
```diff
@@ -0,0 +1,50 @@
+  <project_license>GPL-3.0</project_license>
+  <name>PopcornTime</name>
+  <summary>Watch Movies and TV Shows instantly</summary>
+  ​<translation type="gettext">PopcornTime</translation>
```
Does it actually have translations?

TingPing commented:
OK so in the end I don't believe we will want to host this on Flathub. Promoting piracy is probably not something we want to do. The package seems otherwise functional so you'll just have to host it somewhere else.

aiamuzz commented:
@Preisschild ... so will you be hosting this package outside flathub ?
do share the link to the same if you do ...

Preisschild commented:
@aiamuzz I currently have the Flatpak stuff with instructions on Gitlab: https://gitlab.com/Preisschild/popcorntime-flatpak/blob/master/README.md

I'm planning on opening a Flatpak Repository for Piracy Related stuff like stremio-flatpak and popcorntime-flatpak

TingPing commented:
@Preisschild BTW your docs can be simplified into two commands:

```sh
flatpak remote-add --if-not-exists flathub https://dl.flathub.org/repo/flathub.flatpakrepo
flatpak-builder --install --install-deps-from=flathub popcorntime sh.popcorntime.PopcornTime.yml

TingPing commented:
And you should merge the other reviews here, because currently your appdata file is totally invalid (Always use `appstream-util validate` anyway)

TingPing commented:
> I did that an hour ago already.

I left more comments after.

> appstream-util validate got me an "OK" as well.

That is disappointing, it should detect obviously wrong things =\

Preisschild commented:
@TingPing I'm blind, didn't see that review of the launchable first. Now its merged. Thank you!
